subsys/net/lib/coap/coap_client.c

```diff
@@ -541,30 +541,32 @@ static bool token_compare(struct coap_client_internal_request *internal_req,
 
 static int recv_response(struct coap_client *client, struct coap_packet *response)
 {
-	int len;
+	int total_len;
+	int available_len;
 	int ret;
 
 	memset(client->recv_buf, 0, sizeof(client->recv_buf));
-	len = receive(client->fd, client->recv_buf, sizeof(client->recv_buf), ZSOCK_MSG_DONTWAIT,
-		      &client->address, &client->socklen);
+	total_len = receive(client->fd, client->recv_buf, sizeof(client->recv_buf),
+			ZSOCK_MSG_DONTWAIT | ZSOCK_MSG_TRUNC, &client->address, &client->socklen);
 
-	if (len < 0) {
+	if (total_len < 0) {
 		LOG_ERR("Error reading response: %d", errno);
 		return -EINVAL;
-	} else if (len == 0) {
+	} else if (total_len == 0) {
 		LOG_ERR("Zero length recv");
 		return -EINVAL;
 	}
 
-	LOG_DBG("Received %d bytes", len);
+	available_len = MIN(total_len, MAX_COAP_MSG_LEN);
```

Comment:
Technically the same, but I'd be easier to follow if `sizeof(client->recv_buf)` was used here.